### PR TITLE
C125-148: Form control number field localization

### PR DESCRIPTION
**Pull request by dsuren1**
## Description
Allow input type=number field to display the number based on locale to denote the decimalSeparator and thousandSeparator. 

**Please check if the PR fulfills these requirements**
- [x] The commit message follows our guidelines: https://github.com/geosolutions-it/MapStore2/blob/master/CONTRIBUTING.md
- [x] Tests for the changes have been added (for bug fixes / features)
- [ ] Docs have been added / updated (for bug fixes / features)

**What kind of change does this PR introduce?** (check one with "x", remove the others)
 - [ ] Bugfix
 - [x] Feature
 - [ ] Code style update (formatting, local variables)
 - [ ] Refactoring (no functional changes, no api changes)
 - [ ] Build related changes
 - [ ] CI related changes
 - [ ] Other... Please describe:

## Issue
**What is the current behavior?**
#5499 
Input field of type=number doesn't display locale-based result

**What is the new behavior?**
New custom form control of type number will have the following feature while displaying the number in localized format 

- Formatting and parsing (both in the current language)
- step (arrows, like in the original number type)
- Handle type (integer, decimal) and limits, number of digits... (min/max)

## Breaking change
**Does this PR introduce a breaking change?** (check one with "x", remove the other)
 - [ ] Yes, and I documented them in migration notes
 - [x] No

## Other useful information
Refer: https://github.com/geosolutions-it/austrocontrol-C125/issues/148

### CODE REVIEW

**Comment by dsuren1** [comment]
Hi @offtherailz 
Kindly check my sandbox example implementing,
- Formatting and parsing (both in the current language)
- step (arrows, like in the original number type)
- Handle type (integer, decimal) and limits, number of digits... (min/max)

with the combination of **[react-numeric-input](https://www.npmjs.com/package/react-numeric-input)** library for input field and format/parse with @brightspace-ui/intl library.
Best thing about react-numeric-input it allows both format and parse while maintaining the input field as number with spinners
https://codesandbox.io/s/wispy-paper-c86lo?file=/src/App.js

**Comment by offtherailz** [comment]
Both the implementation are good. To avoid to include a dependency only for this purpose i should proceed for the native implementation

You can at anytime replace the NumberParser implementation with the `brightspace-ui/intl` one in the future (if you need more features or to avoid maintainence) i think in a transparent way, 
 Make you sure to keep :
- the existing options (maxNumberOfDigits ...) available and all properties and handlers as the normal input type="number"
- Check how does it fit with react-bootstrap (error states for instance) and FeatureEditor inputs NumberEditor

**Review by offtherailz — CHANGES_REQUESTED** [review]
_note: I did this test aligning the branch with master
- [x] When I edit numeric field in feature grid (open feature grid on state, turn on edit mode, double click on numeric field) I hav ethis error and Ops screen
```
webpack:///./web/client/components/app/StandardRouter.jsx?:135 TypeError: inputNode.focus is not a function
    at EditorContainer.setTextInputFocus (webpack:///./node_modules/react-data-grid/dist/react-data-grid.js?:3804)
    at EditorContainer.componentDidMount (webpack:///./node_modules/react-data-grid/dist/react-data-grid.js?:3454)
    at commitLifeCycles (webpack:///./node_modules/react-dom/cjs/react-dom.development.js?:21964)
    at commitLayoutEffects (webpack:///./node_modules/react-dom/cjs/react-dom.development.js?:25288)
    at HTMLUnknownElement.callCallback (webpack:///./node_modules/react-dom/cjs/react-dom.development.js?:362)
    at Object.invokeGuardedCallbackDev (webpack:///./node_modules/react-dom/cjs/react-dom.development.js?:411)
    at invokeGuardedCallback (webpack:///./node_modules/react-dom/cjs/react-dom.development.js?:466)
    at commitRootImpl (webpack:///./node_modules/react-dom/cjs/react-dom.development.js?:25024)
    at unstable_runWithPriority (webpack:///./node_modules/scheduler/cjs/scheduler.development.js?:815)
    at runWithPriority$2 (webpack:///./node_modules/react-dom/cjs/react-dom.development.js?:12188)
    at commitRoot (webpack:///./node_modules/react-dom/cjs/react-dom.development.js?:24865)
    at finishSyncRender (webpack:///./node_modules/react-dom/cjs/react-dom.development.js?:24251)
    at performSyncWorkOnRoot (webpack:///./node_modules/react-dom/cjs/react-dom.development.js?:24223)
    at eval (webpack:///./node_modules/react-dom/cjs/react-dom.development.js?:12238)
    at unstable_runWithPriority (webpack:///./node_modules/scheduler/cjs/scheduler.development.js?:815)
    at runWithPriority$2 (webpack:///./node_modules/react-dom/cjs/react-dom.development.js?:12188)
onError @ webpack:///./web/client/components/app/StandardRouter.jsx?:135
```
- [x] Maybe is a good idea to format also the numbers in table with a formatter in the table with the same style (I se 12.34 instead of 12,34 in italian but when I edit I could see 12,34 that is misleading)
- [x] As wel as I try to edit geometry details in query panel (North input), I have this error, ops screen:
```
react_devtools_backend.js:2273 TypeError: Cannot read property 'value' of undefined
    at Object.onChange (GeometryDetails.jsx:187)
    at onChange (IntlNumberFormControl.jsx:141)
    at NumericInput._invokeEventCallback (index.js:469)
    at NumericInput.componentDidUpdate (index.js:179)
    at commitLifeCycles (react-dom.development.js:21980)
    at commitLayoutEffects (react-dom.development.js:25288)
    at HTMLUnknownElement.callCallback (react-dom.development.js:362)
    at Object.invokeGuardedCallbackDev (react-dom.development.js:411)
    at invokeGuardedCallback (react-dom.development.js:466)
    at commitRootImpl (react-dom.development.js:25024)
    at unstable_runWithPriority (scheduler.development.js:815)
    at runWithPriority$2 (react-dom.development.js:12188)
    at commitRoot (react-dom.development.js:24865)
    at finishSyncRender (react-dom.development.js:24251)
    at performSyncWorkOnRoot (react-dom.development.js:24223)
    at eval (react-dom.development.js:12238) "
    at ErrorBoundary (webpack:///./node_modules/react-error-boundary/dist/commonjs/ErrorBoundary.js?:36:5)
    at div
    at Router (webpack:///./node_modules/react-router/es/Router.js?:33:5)
    at ConnectedRouter (webpack:///./node_modules/connected-react-router/esm/ConnectedRouter.js?:56:7)
    at ConnectedRouterWithContext (webpack:///./node_modules/connected-react-router/esm/ConnectedRouter.js?:154:25)
    at Connect(ConnectedRouterWithContext) (webpack:///./node_modules/react-redux/es/components/connectAdvanced.js?:151:37)
    at IntlProvider (webpack:///./node_modules/react-intl/lib/index.es.js?:802:9)
    at Localized (webpack:///./web/client/components/I18N/Localized.jsx?:52:5)
    at Theme (webpack:///./web/client/components/theme/Theme.jsx?:108:5)
    at SideEffect(Theme) (webpack:///./node_modules/react-side-effect/lib/index.js?:62:9)
    at Connect(SideEffect(Theme)) (webpack:///./node_modules/react-redux/es/components/connectAdvanced.js?:151:37)
    at div
    at StandardRouter (webpack:///./web/client/components/app/StandardRouter.jsx?:88:5)
    at Connect(StandardRouter) (webpack:///./node_modules/react-redux/es/components/connectAdvanced.js?:151:37)
    at DragDropContext(Connect(StandardRouter)) (webpack:///./node_modules/react-dnd/lib/DragDropContext.js?:81:5)
    at Provider (webpack:///./node_modules/react-redux/es/components/Provider.js?:21:24)
    at StandardApp (webpack:///./web/client/components/app/StandardApp.jsx?:132:5)"
```
- [x] Annotations editor hides last digit of seconds. Maybe this will need a re-check with this pending PR:
https://github.com/geosolutions-it/MapStore2/pull/5776
image
- [x] Timeline settings: some issues with style:
image
- [ ] Legend input in Print options should be only integer, but allows to insert decimals
maybe inside the new styler
- [ ] Please prefer, where possible css precendence rules to !important if you are not overriding existing libraries

- [ ] I Didn't tested 
    - [ ] Measure editor
    - [ ] Dwithin editor (is it used anymore?)
    - [ ] Rules editor
    - [ ] components/style/TextStyler and Text (does it exists or is used anymore?) @allyoucanmap 
For these, we need to check if everything is ok, and / or if they can be removed.

File: web/client/components/I18N/IntlNumberFormControl.jsx [hunk 1]
```diff
@@ -0,0 +1,103 @@
...
+ *
+ * This source code is licensed under the BSD-style license found in the
+ * LICENSE file in the root directory of this source tree.
+ */
+const PropTypes = require('prop-types');
+const React = require('react');
+require('./css/formControlIntl.css');
+const NumericInput = require("react-numeric-input");
+
+class IntlNumberFormControl extends React.Component {
```
Comment (offtherailz) [inline web/client/components/I18N/IntlNumberFormControl.jsx:13]:
```suggestion
/**
 * Localized Numeric Input. It provides an numeric input value that uses separators (eg. ",",".") as they
 * are used in the current selected language.
 * @prop {string|number} value a numeric value
 * @prop {string|number] defaultValue a value to use as default
 * @prop {function} onChange handler of change event. the argument of this handler is the current value inserted.
 * @prop {number} step the step when you click on arrows (up, down) of the input
 * @prop {function} onBlur event handler on blur event
 */
class IntlNumberFormControl extends React.Component {
```

File: web/client/components/I18N/IntlNumberFormControl.jsx [hunk 2]
```diff
@@ -0,0 +1,111 @@
+/**
```
Comment (offtherailz) [inline web/client/components/I18N/IntlNumberFormControl.jsx:1]:
```suggestion
/*
```

**Review by offtherailz — APPROVED** [review]

**Pull request merged**